### PR TITLE
Writing Flow: fix empty horizontal arrow nav

### LINKED ISSUE #16841: WritingFlow: Arrow keys occasionally error, do not navigate to next block
**Describe the bug**

In writing, I have found that sometimes pressing <kbd>ArrowLeft</kbd> or <kbd>ArrowRight</kbd> does not navigate to the previous or next blocks as I would expect them too. There is also an error logged to the console:

```
react-dom.165d5c53.js:283 Uncaught TypeError: Cannot read property 'right' of undefined
    at isEdge (dom.js:166)
    at isHorizontalEdge (dom.js:178)
    at WritingFlow.onKeyDown (index.js:24422)
    at HTMLUnknownElement.callCallback (react-dom.165d5c53.js:143)
    at Object.invokeGuardedCallbackDev (react-dom.165d5c53.js:193)
    at invokeGuardedCallback (react-dom.165d5c53.js:250)
    at invokeGuardedCallbackAndCatchFirstError (react-dom.165d5c53.js:264)
    at executeDispatch (react-dom.165d5c53.js:555)
    at executeDispatchesInOrder (react-dom.165d5c53.js:574)
    at executeDispatchesAndRelease (react-dom.165d5c53.js:674)
```

https://github.com/WordPress/gutenberg/blob/cd3baf4d9999886b7b7a3add730456d56b75c373/packages/dom/src/dom.js#L166

(last git blame #14462, cc @ellatrix @nosolosw)

**To reproduce**

1. Navigate to Posts > Add New
2. Click the writing prompt
3. Press <kbd>Enter</kbd>
4. Press <kbd>ArrowLeft</kbd>
5. Press <kbd>ArrowRight</kbd>
4. Observe the first block is still selected, an error logged to the console.

**Expected behavior**

The second block should be selected, and no error should be logged.

**Desktop (please complete the following information):**

 - OS: macOS Mojave 10.14.5 (18F203)
 - Browser: Chrome Version 75.0.3770.100 (Official Build) (64-bit)

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #16841.

It seems like the cause of these regressions are:

1. The rich text element no longer being relatively positioned broke the z-index trick to correctly calculate a range.
2. The new placeholder (#16733) seems to create a selection that is sometimes:
    * Uncollapsed when the user clicks in an empty paragraph (even though it looks collapsed).
    * Sometimes a bit further away, but still no more than 1 pixel.

Solution:

1. Always position the element relatively when using the z-index trick.
2. Set `contenteditable=false` on the placeholder so that it never catches selection.
3. Increase the buffer for checking if horizontal selection is at the edge from about <=0.5 px to 1px.

## How has this been tested?
See #16841.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Comment by ellatrix** [comment]
Rebased. This regression is important to be fixed ASAP I think. :) Some of it has been introduced by #16733.

File: packages/rich-text/src/to-tree.js [hunk 1]
```diff
@@ -255,6 +255,7 @@ export function toTree( {
 					type: 'span',
 					attributes: {
 						'data-rich-text-placeholder': placeholder,
+						contenteditable: 'false',
```
Comment (talldan) [inline packages/rich-text/src/to-tree.js]: I don't know this part of the codebase particularly well, how does this change relate to the bug?

**Review by talldan — APPROVED** [review]
I noticed this for the first time today, so it's great there's a fix ready! It worked well in my testing. I'm not completely familiar with the code or how `isEdge` works, but when debugging I noticed the issue was with the position comparison returning false, so the fix seems good.

File: packages/rich-text/src/to-tree.js (hunk 1 above)
Comment (ellatrix) [reply packages/rich-text/src/to-tree.js]: Without this attribute, the placeholder would catch selection, with it, it cannot. I'll leave a comment.

**Comment by ellatrix** [comment]
Thanks for the review @talldan!

**Pull request merged**